packages/sanes-chrome-extension/src/routes/share-identity/index.tsx

```diff
@@ -1,19 +1,32 @@
+import { ToastContext } from 'medulas-react-components/lib/context/ToastProvider';
 import * as React from 'react';
+import { RouteComponentProps, withRouter } from 'react-router';
+import { RequestContext } from '../../context/RequestProvider';
+import { history } from '../../store/reducers';
+import { REQUEST_ROUTE } from '../paths';
+import { checkRequest } from '../requests';
 import RejectRequest from './components/RejectRequest';
 import ShowRequest from './components/ShowRequest';
 
-const ShareIdentity = (): JSX.Element => {
+const ShareIdentity = ({ location }: RouteComponentProps): JSX.Element => {
   const [action, setAction] = React.useState<'show' | 'reject'>('show');
+  const toast = React.useContext(ToastContext);
+  const requestContext = React.useContext(RequestContext);
+
+  const request = requestContext.firstRequest;
```

Comment:
Suggestion: you can write `const request = requestContext.requests.find(() => true);`, which looks a bit odd but gives you the first element if there is one. This allows you to avoid the duplicated data.

Reply:
I was going to modify thinking here I was using `requests` and `firstRequest`, but I realised only the `firstRequest` is used. I prefer to keep it in this way because it explains better internal requirements and it is encapsulated correctly.